.github/workflows/leverage-cli-test.yml

```diff
@@ -0,0 +1,59 @@
+name: Leverage CLI Test
+
+on:
+  workflow_dispatch:
+    inputs:
+      leverage_branch:
+        description: "Reference to current Leverage branch to test"
+        required: true
+        type: string
+
+jobs:
+  test_leverage:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout base branch
+        uses: actions/checkout@v3
+
+      - name: Clone Leverage
+        run: |
+          printf "[INFO] Cloning Leverage CLI's repo"
+          git clone git@github.com:binbashar/leverage.git
+
+      - name: Build Leverage CLI
+        run: |
+          printf "[INFO] Building Leverage CLI"
+          git checkout ${{ inputs.leverage_branch }}
+          make build
+          pip install -e .
+        working-directory: ./leverage
+
+      - name: Set up credentials
```

Comment:
@angelofenoglio I add an alternative for this step

[ ] ["Configure AWS Credentials" Action For GitHub Actions](https://github.com/aws-actions/configure-aws-credentials)

Reply:
That action looks quite nice, actually. But should we trust credentials handling to a third party?

CC: @exequielrafaela @diego-ojeda-binbash

Reply:
I'll go ahead and merge this for now just to make some progress on the workflows. Since we'll most certainly be making changes along the way we can easily address this within those changes.

Reply:
@angelofenoglio Good concern. Although the action @marianod92 recommended comes from AWS. If we still do not trust it we could work on a fork. Now, that said, let's keep that action in our radar for later.

Reply:
@angelofenoglio I've noted what @diego-ojeda-binbash commented, since this is the official AWS GH action sounds like a very good and safe option. However since this actions directly manage our Leverage Reference Architecture  accounts which are very sensitive I would rather move to self-hosted runners in the short term to avoid any chances of credentials exposure in our GH workflows.

CC: @marianod92 @juanmatias @binbashar/leverage-ref-architecture-aws-admin @binbashar/leverage-ref-architecture-aws-dev